cloud/dimensiondata/dimensiondata_vlan.py

```diff
@@ -0,0 +1,411 @@
+            vlan = driver.ex_get_vlan(vlan_id)
+            return vlan
+        except DimensionDataAPIException as e:
+            if e.code == "NOT_FOUND":
+                return {'id': False}
+            else:
+                module.fail_json(msg="Unexpected error: %s" % e)
+    elif name is not False:
+        try:
+            # Find vlan by name
+            vlans = driver.ex_list_vlans(location, network_domain, name)
+            # VLAN not found
+            if len(vlans) < 1:
+                return {'id': False}
+            return driver.ex_get_vlan(vlans[0].id)
+        except DimensionDataAPIException as e:
+            module.fail_json(msg="Unexpected API error: %s" % e)
+    else:
+        module.fail_json(msg="Unexpected Error: One of vlan_id or name " +
+                             "must be provided.")
+
+
+def delete_vlan(module, driver, location, network_domain, vlan_id, name):
+    vlan = get_vlan(module, driver, location, network_domain,
+                    vlan_id, name)
+    if type(vlan) is dict:
+        return False
+    try:
+        return driver.ex_delete_vlan(vlan)
+    except DimensionDataAPIException as e:
+        module.fail_json(msg="Unexpected API error: %s" % e)
+
+
+def update_vlan(module, driver, location, network_domain, vlan_id,
+                name, description):
+    vlan_obj = get_vlan(module, driver, location, network_domain,
+                        vlan_id, name)
+    if vlan_obj is False:
+        changed = False
+        vlan = False
+    elif vlan_obj.name != name or vlan_obj.description != description:
+        vlan_obj.name = name
+        vlan_obj.description = description
+        try:
+            vlan = driver.ex_update_vlan(vlan_obj)
+        except DimensionDataAPIException as e:
+            module.fail_json(msg="Failed to update VLAN: %s" % e)
+        changed = True
+    else:
+        changed = False
+        vlan = False
+    return {'changed': changed, 'vlan': vlan}
+
+
+def expand_vlan(module, driver, location, network_domain, vlan_id, name,
+                private_ipv4_range_size):
+    vlan_obj = get_vlan(module, driver, location, network_domain,
+                        vlan_id, name)
+    if type(vlan_obj) is dict:
+        module.exit_json(changed=False, msg="VLAN not found.")
+    else:
+        if vlan_obj.private_ipv4_range_size == private_ipv4_range_size:
+            return False
+        else:
+            vlan_obj.private_ipv4_range_size = private_ipv4_range_size
+            try:
+                return driver.ex_expand_vlan(vlan_obj)
+            except DimensionDataAPIException as e:
+                module.fail_json(msg="Failed to expand VLAN: %s" % e)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            region=dict(default='na', choices=dd_regions),
+            location=dict(required=True, type='str'),
+            network_domain=dict(required=True, type='str'),
+            name=dict(default=False, type='str'),
```

Comment:
Note -- the default of the default parameter is None.  That's usually a better sentinel value than speficying a boolean here.